src/meta/src/hummock/manager/commit_epoch.rs

```diff
@@ -189,6 +219,45 @@ impl HummockManager {
             }
         }
 
+        let batch_commit_for_new_cg = if !batch_commit_for_new_cg.is_empty() {
+            let mut new_id_count = 0;
+            let mut batch_commit_info = HashMap::new();
+            for BatchCommitForNewCg {
+                epoch_to_ssts,
+                table_ids,
+            } in batch_commit_for_new_cg
+            {
+                let new_compaction_group_id = next_compaction_group_id(&self.env).await?;
+                compaction_group_manager.insert(
+                    new_compaction_group_id,
+                    CompactionGroup {
+                        group_id: new_compaction_group_id,
+                        compaction_config: compaction_group_config.clone(),
+                    },
+                );
+
+                new_id_count += epoch_to_ssts.values().map(|ssts| ssts.len()).sum::<usize>();
+
+                on_handle_add_new_table(
+                    &state_table_info,
+                    &table_ids,
+                    new_compaction_group_id,
+                    &mut table_compaction_group_mapping,
+                    &mut new_table_ids,
+                )?;
+
+                batch_commit_info.insert(new_compaction_group_id, epoch_to_ssts);
+            }
+            let start_sst_id = next_sstable_object_id(&self.env, new_id_count).await?;
```

Comment:
I think we don't need to reserve new sst ids for the new compaction group, because its parent group will be StaticCompactionGroupId::NewCompactionGroup so that the reserved new sst ids won't never used anyway.

Reply:
In this `commit_epoch`, we will insert data to the newly created compaction together in a single version delta, so we may need to reserve some sst ids for the newly added data?

Reply:
The newly added SSTs already have their deterministic SST id.
We only need reserve SST ids in advance when branching SST by splitting compaction group.

Reply:
Get it. I just check the usage of `new_sst_start_id `, and indeed there is no need to set it when we create an empty new compaction group. Have changed to avoid passing this `new_sst_start_id ` value along the way.